backend.native/tests/runtime/workers/worker10.kt

```diff
@@ -78,4 +78,24 @@ val topSharedData = Data(43)
 
     worker.requestTermination().result
     println("OK")
+}
+
+val ref = AtomicReference<Any?>(Any().freeze())
+
+val semaphore = AtomicInt(0)
+
+@Test fun runTest2() {
```

Comment:
This test doesn't reproduce [the original bug](https://github.com/JetBrains/kotlin-native/pull/2912#discussion_r278066070) on [cb7e81a7](https://github.com/JetBrains/kotlin-native/commit/cb7e81a7e20f9efcd81cf4a94660e488da933324).

Reply:
Hopefully now it does.